contrib/kind-common

```diff
@@ -0,0 +1,369 @@
+  fi
+  delete_metallb_dir
+}
+
+delete_metallb_dir() {
+  if ! [ -d "${METALLB_DIR}" ]; then
+      return
+  fi
+
+  # The build directory will contain read only directories after building. Files cannot be deleted, even by the owner.
+  # Therefore, set all dirs to u+rwx.
+  find "${METALLB_DIR}" -type d -exec chmod u+rwx "{}" \;
+  rm -rf "${METALLB_DIR}"
+}
+
+# kubectl_wait_pods will set a total timeout of 300s for IPv4 and 480s for IPv6. It will first wait for all
+# DaemonSets to complete with kubectl rollout. This command will block until all pods of the DS are actually up.
+# Next, it iterates over all pods with name=ovnkube-db and ovnkube-master and waits for them to post "Ready".
+# Last, it will do the same with all pods in the kube-system namespace.
+kubectl_wait_pods() {
+  # IPv6 cluster seems to take a little longer to come up, so extend the wait time.
+  OVN_TIMEOUT=300
+  if [ "$KIND_IPV6_SUPPORT" == true ]; then
+    OVN_TIMEOUT=480
+  fi
+
+  # We will make sure that we timeout all commands at current seconds + the desired timeout.
+  endtime=$(( SECONDS + OVN_TIMEOUT ))
+
+  for ds in ovnkube-node ovs-node; do
+    timeout=$(calculate_timeout ${endtime})
+    echo "Waiting for k8s to launch all ${ds} pods (timeout ${timeout})..."
+    kubectl rollout status daemonset -n ovn-kubernetes ${ds} --timeout ${timeout}s
+  done
+
+  pods=""
+  if [ "$OVN_ENABLE_INTERCONNECT" == true ]; then
+    pods="ovnkube-control-plane"
+  else
+    pods="ovnkube-master ovnkube-db"
+  fi
+  for name in ${pods}; do
+    timeout=$(calculate_timeout ${endtime})
+    echo "Waiting for k8s to create ${name} pods (timeout ${timeout})..."
+    kubectl wait pods -n ovn-kubernetes -l name=${name} --for condition=Ready --timeout=${timeout}s
+  done
+
+  timeout=$(calculate_timeout ${endtime})
+  if ! kubectl wait -n kube-system --for=condition=ready pods --all --timeout=${timeout}s ; then
+    echo "some pods in the system are not running"
+    kubectl get pods -A -o wide || true
+    exit 1
+  fi
+}
+
+# calculate_timeout takes an absolute endtime in seconds (based on bash script runtime, see
+# variable $SECONDS) and calculates a relative timeout value. Should the calculated timeout
+# be <= 0, return one second.
+calculate_timeout() {
+  endtime=$1
+  timeout=$(( endtime - SECONDS ))
+  if [ ${timeout} -le 0 ]; then
+      timeout=1
+  fi
+  echo ${timeout}
+}
+
+sleep_until_pods_settle() {
+  echo "Pods are all up, allowing things settle for 30 seconds..."
+  sleep 30
+}
+
+function is_nested_virt_enabled() {
+    local kvm_nested="unknown"
+    if [ -f "/sys/module/kvm_intel/parameters/nested" ]; then
+        kvm_nested=$( cat /sys/module/kvm_intel/parameters/nested )
+    elif [ -f "/sys/module/kvm_amd/parameters/nested" ]; then
+        kvm_nested=$( cat /sys/module/kvm_amd/parameters/nested )
+    fi
+    [ "$kvm_nested" == "1" ] || [ "$kvm_nested" == "Y" ] || [ "$kvm_nested" == "y" ]
+}
+
+function install_kubevirt() {
```

Comment:
Note: inconsistent function format, but decided to leave it unchanged to make review easier to chec